DOC: Fix typos and formatting in .rst files and Python examples.

jhlegarreta opened the pull request:
Fix typos and formatting in .rst files and Python examples'
documentations:
- Fix typos in files.
- Replace tabs for two white spaces.
- Fix errors in citation and DIPY internal cross-refs.
- Add missing cross-refs (e.g. Aganj MRM 2010).
- Add links where necessary (ISBI HARDI Contest 2013 and FreeSurfer).
- Fix LaTeX math formatting errors.
- Make the example/DIPY code object markdown consistent (inverted commas).
- Capitalize the acronym long names' first/corresponding letters (e.g.
  Constrained Spherical Deconvolution).
- Capitalize acronyms (e.g. FA).
- Make the writing of terms consistent in terms of uppercase/lowercase
  (e.g. b0 vs. B0; cospus callosum vs. Corpus Callosum, etc.).
- Use the [NameYear] convention for citations.
- Place all citations under a 'References' section, and use the subsection
  markdown for the section.
- Use inverted commas consistently to reference code objects.
- Make the b-value units consistent across files (s/mm^2).
- Create and include missing figures for 'reconst_fwdti.py'.
- Write DIPY instead of any of its variants in the documentation.
- Link the first appearance of dipy in every file (use dipy_).

jhlegarreta commented:
This PR addresses the issues in issue #1305.

Folks, sorry for the delay, but have been working on this only intermittently.

Thanks to all of you who made comments.

I think I've covered all I had written down in my notes. Surely I did not cover all files, nor all markdown (i.e. inverted commas for DIPY objects), so please feel free to request a revision to the PR.

I guess we'll be able to see whether the cross-refs work/have been fixed properly (e.g. cross-refs between files) once this gets merged.

I'll work on the the guidelines on a separate topic. 

@garyfallidis I tried to use DIPY systematically in the docs. I linked its first appearance with dipy_. Have also encountered `dipy_`. They both seemed to be linked correctly. Please, let me know which is preferred.

A few issues still remain:

- @skoudoro , have been looking at it, but could not find a solution; the same issue is now (the previous version only cross-ref'ed the first appearance) present in http://nipy.org/dipy/examples_built/reconst_dki.html#example-reconst-dki

However, I guess that is the way the Sphinx guys thought it should work; when clicking on the number we can go back to the location of the reference.

I guess this may be cumbersome when the same reference is used more than, say, three times in the same file.

- In the `restore_dti.py` example the third reference not used: should we remove it? Have gone through the doc, but haven't found the missing location. Anyone knowing where it fits?

- In the `snr_in_cc.py` file which is the right link for this?
(see the DTI example for further explanations).

- Is this file linked or listed anywhere?

> simulate_dki.py

Should I include it in the Simulation section of `examples_index.rst`?

- In the `fiber_to_bundle_coherence.py` example Fig. 1 is not labeled, so the first sentence is missing it.

- The figures for the `reconst_fwdti.py` example are not being rendered; they are produced correctly when running the example, and have seen that the `/tools/make_examples.py` would compile and produce all figures to put them in `/doc/examples_built/_static`, but that does not seem to be happening for the above figures. The file at issue is listed in `examples_index.rst`, so I'm not seeing the problem. Anyone?

skoudoro commented:
Awesome ! Thanks again ! I will review and answer yours questions as soon as possible

skoudoro commented:
> In the restore_dti.py example the third reference not used: should we remove it? Have gone through the doc, but haven't found the missing location. Anyone knowing where it fits?

it fits at the end of this sentence : `using the RESTORE estimate and the WLS estimate`

> In the snr_in_cc.py file which is the right link for this?

I do not understand what you mean here

> Should I include it in the Simulation section of examples_index.rst?

yes, thanks.

> In the fiber_to_bundle_coherence.py example Fig. 1 is not labeled, so the first sentence is missing it.

It seems that all description below the figures are not labeled so you can define them.

> The figures for the reconst_fwdti.py example are not being rendered;  ....

I think, we should create a specific issue for this one.  this deserve a new PR like #1301

jhlegarreta commented:
@skoudoro thanks for the review ! Commit 8c7b3d3 addresses the comments. I've opened the issue  #1317 to track the `reconst_fwdti.py` example figures not being rendered. 

The figures should now be labeled and referenced in `fiber_to_bundle_coherence.py`. However, if the entire caption is set as the link text, it will be way too long for the first figure. Any thoughts? 

By,
``In the snr_in_cc.py file which is the right link for this?
(see the DTI example for further explanations).``

I meant that the `snr_in_cc.py` file contains the mentioned sentence, without actually linking to any specific example, and I was wondering which file it should be linked to.

skoudoro commented:
oh ok, I see, it should be link on [this example](http://nipy.org/dipy/examples_built/reconst_dti.html#example-reconst-dti) (example_reconst_dti.py)

> In the last figure of the same file, I also added the :math: syntax in an attempt to render in math syntax the threshold. Not sure whether that will work.

I'm generating the whole documentation with your change. I will let you know.

jhlegarreta commented:
@skoudoro Done for the link of `reconst_dti.py` when cross-ref'ing in `snr_in_cc.py`.

jhlegarreta commented:
@skoudoro Thanks for having a look to the doc generation ! Please, pay attention to some links; I'm not sure whether Sphinx allows to break the link text or link text vs. link across lines, which I've done.

Also, across the website some paths are written as `<./doc/devel>` while others are simply formatted as `doc/devel`. Just a minor detail, but for the sake of consistency, which one should we encourage?

skoudoro commented on doc/examples/combined_workflow_creation.py:
```diff
@@ -3,13 +3,13 @@
 
-A combined workflow is a series of dipy workflows organized together in a way
-that the output of a workflow serves as input for the next one.
+A ``CombinedWorkflow`` is a series of dipy_ workflows organized together in a
```
`dipy_` link does not work. it refers to `#id1`

jhlegarreta replied:
Same as above.

skoudoro commented on doc/examples/fiber_to_bundle_coherence.py:
```diff
@@ -30,14 +32,14 @@
-(150 orientations, b=2000s/mm^2) which is one of the standard example datasets
-in DIPY.
+(150 orientations, b=2000 $s/mm^2$) which is one of the standard example
+datasets in dipy_.
```
same as above, this link does not work

jhlegarreta replied:
Same as above.

skoudoro commented on doc/examples/fiber_to_bundle_coherence.py:
```diff
@@ -6,9 +6,11 @@
+[Meesters2016]_. These measures are useful in “cleaning” the results of
 tractography algorithms, since low FBCs indicate which fibers are isolated and
-poorly aligned with their neighbors, see Fig. 1.
+poorly aligned with their neighbors, see :ref:`fiber_to_bundle_coherence`.
```
problem here, the whole caption appears

jhlegarreta replied:
I assumed splitting the paragraph would not solve the issue, so I just referenced the figure below in 412e9fb. I'm open to suggestions.

skoudoro commented on doc/examples/gradients_spheres.py:
```diff
@@ -3,22 +3,22 @@
-This example shows how you can create GradientTables and Sphere objects using 
-Dipy.
+This example shows how you can create gradient tables and sphere objects using
+dipy_.
```
the link is working here :-)

jhlegarreta replied:
I guess it's because the `link_names.inc` file was included.

skoudoro commented on doc/examples/gradients_spheres.py:
```diff
@@ -177,10 +178,16 @@
+   anisotropic systems by magnetic resonance imaging, Magnetic Resonance in
+   Medicine, vol 42, no 3, 515-525, 1999.
+
+.. include:: ../links_names.inc
```
`.. admonition` does not work here

jhlegarreta replied:
Haven't got an explanation for this.

skoudoro commented on doc/examples/kfold_xval.py:
```diff
@@ -22,7 +22,7 @@
 compare models that are not nested.
 
-In `dipy`, we include an implementation of k-fold cross-validation. In this
+In dipy_, we include an implementation of k-fold cross-validation. In this
```
no problem here too

jhlegarreta replied:
I guess it's because the `link_names.inc` file was included.

skoudoro commented on doc/examples/segment_extending_clustering_framework.py:
```diff
@@ -7,47 +7,51 @@
 Clustering framework
 ====================
-Dipy provides a simple, flexible and fast framework to do clustering of
+dipy_ provides a simple, flexible and fast framework to do clustering of
```
link does not work

jhlegarreta replied:
I guess it's because the `link_names.inc` file was NOT included. Fixed in 412e9fb.

skoudoro commented on doc/examples/simulate_multi_tensor.py:
```diff
@@ -76,7 +76,7 @@
 For the ODF simulation we will need a sphere. Because we are interested in a
 simulation of only a single voxel, we can use a sphere with very high
-resolution. We generate that by subdividing the triangles of one of Dipy's
+resolution. We generate that by subdividing the triangles of one of dipy_'s
```
link does not work

jhlegarreta replied:
I guess it's because the `link_names.inc` file was NOT included. Fixed in 412e9fb.

skoudoro commented on doc/examples/streamline_length.py:
```diff
@@ -7,7 +7,7 @@
 lengths or overall shape.
 
-A streamline in Dipy is represented as a numpy array of size
+A streamline in dipy_ is represented as a numpy array of size
```
same as above

jhlegarreta replied:
Same as above.

skoudoro commented on doc/examples/viz_advanced.py:
```diff
@@ -3,7 +3,7 @@
 ==================================
 
-In DIPY we created a thin interface to access many of the capabilities
+In dipy_ we created a thin interface to access many of the capabilities
```
same as above

skoudoro commented on doc/examples/viz_surfaces.py:
```diff
@@ -3,19 +3,19 @@
 
-Here is a simple tutorial that shows how to visualize surfaces using DIPY. It
-also shows how to load/save, get/set and update vtkPolyData and show
+Here is a simple tutorial that shows how to visualize surfaces using dipy_. It
```
same as above

jhlegarreta replied:
Same as above.

skoudoro commented on doc/examples/workflow_creation.py:
```diff
@@ -3,16 +3,18 @@
 ============================================================
 
-A workflow is a series of dipy operations with fixed inputs and outputs
+A workflow is a series of dipy_ operations with fixed inputs and outputs
```
same as above

jhlegarreta replied:
Same as above.

skoudoro commented on doc/examples/workflow_creation.py:
```diff
@@ -3,16 +3,18 @@
 that is callable via command line or another interface.
 
-For example, after installing dipy, you can call anywhere from your command line:
+For example, after installing dipy_, you can call anywhere from your command
```
same as above

jhlegarreta replied:
Same as above.

skoudoro commented on doc/examples_index.rst:
```diff
@@ -197,6 +197,7 @@ Simulations
 
 - :ref:`example_simulate_multi_tensor`
 - :ref:`example_reconst_dsid`
+- :ref:`simulate_dki`
```
should be `example_simulate_dki` instead of `simulate_dki`

jhlegarreta replied:
Thanks. Fixed in 412e9fb.

skoudoro commented on doc/mission.rst:
```diff
@@ -6,7 +6,7 @@
 Mission of Statement
 
-The purpose of dipy is to make it **easier to do better diffusion MR imaging research**. Following up with the nipy mission statement we aim to build software that is
+The purpose of dipy_ is to make it **easier to do better diffusion MR imaging research**. Following up with the nipy mission statement we aim to build software that is
```
same as above

jhlegarreta replied:
Same as above.

jhlegarreta commented on doc/devel/make_release.rst:
```diff
@@ -75,7 +75,7 @@ Release checklist
-* Check all the dipy builds are green on the `nipy buildbot`_
+* Check all the DIPY builds are green on the `nipy buildbot`_
 
 * If you have travis-ci_ building set up you might want to push the code in its
```
@skoudoro Thanks for giving it a try!

New patch set 412e9fb:

As for the http://nipy.org/dipy/reference_cmd/CombinedWorkflow.html (`combined_workflow_creation.py`) file's strange rendering, I cannot say much; I've gone through it but did not find any error.

As for the files with `.. admonition::`` rendering problems (`combined_workflow_cration.py`, `gradients_spheres.py`), I did not find a plausible explanation.

As for the wrong rendering for `dipy.tracking.life`, I changed to double inverted commas.

As for the links in FAQ, I think I fixed the links for numpy and cython.

Added `travis-ci` to `link-names.inc`. See whether this solves the issue.

jhlegarreta commented on doc/examples/brain_extraction_dwi.py:
```diff
@@ -3,8 +3,8 @@
 
-We show how to extract brain information and mask from a b0 image using dipy's
-segment.mask module.
+We show how to extract brain information and mask from a b0 image using dipy_'s
```
I'd dare to say that the incorrect links to DIPY (`dipy_`) were due to the missing include `links_names.inc`. I've included it in the reported files. See whether this solves the issue.

jhlegarreta commented on doc/devel/make_release.rst:
```diff
@@ -144,11 +144,11 @@ Release checklist
-  At the moment, the useful dipy binary build testers are:
+  At the moment, the useful DIPY binary build testers are:
 
       * http://nipy.bic.berkeley.edu/builders/dipy-bdist32-26
```
Fixed in 412e9fb.

skoudoro commented on doc/examples/simulate_dki.py:
```diff
@@ -24,25 +27,25 @@
 For the simulation we will need a GradientTable with the b-values and
-b-vectors. Here we use the GradientTable of the sample Dipy dataset
-'small_64D'.
+b-vectors. Here we use the GradientTable of the sample dipy_ dataset
```
the link does not work. Missing the include

jhlegarreta commented:
@skoudoro Thanks. Sorry for the delay. I've been away from keyboard until a few days ago. Partially addressed your comments.

> Typo of dipy.reconst.cross_validation on kfold_xval.py

I do not see where the typo is. According to [http://www.sphinx-doc.org/en/stable/domains.html#cross-referencing-python-objects](http://www.sphinx-doc.org/en/stable/domains.html#cross-referencing-python-objects), the markup is fine.

And in fact, that makes me think about changing the statement about using the backquote or inverted commas inline markup for DIPY objects (see the sentence `The classes, objects, and any other construct referenced from the code should be written with inverted commas, (...)` in #1305).

I'd vote for using proper Sphinx cross-referencing/syntax (such as in:mod:`reconst.cross_validation` or its correct form; or ` :class:`GradientTable` which holds all the acquisition specific parameters (...)` instead of ` ``GradientTable`` which holds all the acquisition specific parameters (...)`).

At least for those objects that are present in the dipy code. I ignore whether the rule should also be applied to the objects only present in the example files, or if these should be fine with the inverted commas syntax.

If you do agree, I'd change the above as far as I'm able to in a subsequent patch set.

Also addressed @skoudoro's comments in #1305 concerning the captions and the path syntax. 

Removed remaining tabs, and did some other minor changes for the sake of consistency.

jhlegarreta commented:
The failing build in Travis seems to be unrelated to the patch set.  I ignore the meaning of the `USE_PRE` flag, but I've seen that other builds have the build marked as `Allowed Failures` (e.g. https://travis-ci.org/nipy/dipy/builds/214169250).

arokem commented:
You are seeing https://github.com/nipy/dipy/issues/1323

jhlegarreta commented:
@arokem OK, thanks for clarifying this.

jhlegarreta commented:
Rebased to fix the conflict with master.

skoudoro commented:
Thanks @jhlegarreta !  I merge this one
